extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -91,6 +91,29 @@ public static function onSitemapRobotsPageBeforeOutput( WebRequest $request, Use
 		return true;
 	}
 
+	public static function parserUpgradeVignetteUrls( string &$url ) {
+		if ( preg_match( self::VIGNETTE_IMAGES_HTTP_UPGRADABLE, $url ) && strpos( $url, 'http://' ) === 0 ) {
+			$url = wfHttpToHttps( $url );
+		}
+	}
+
+	/**
+	 * Make sure any "external" links to our own wikis that support HTTPS
+	 * are protocol-relative on output.
+	 *
+	 * @param  string  &$url
+	 * @param  string  &$text
+	 * @param  bool    &$link
+	 * @param  array   &$attribs
+	 * @return boolean
+	 */
+	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
+		if ( wfHttpsAllowedForURL( $url ) ) {
+			$url = wfProtocolUrlToRelative( $url );
+		}
+		return true;
```

Comment:
I am not sure if that return is required. Specially that information doesn't give you anything.

Reply:
It's required because the return is checked by the place the hook is called in order to check if a hook method wants to hijack the output completely: https://github.com/Wikia/app/blob/2fb5243/includes/Linker.php#L1063-L1065

Reply:
Right... Then why this functions doesn't have it? https://github.com/Wikia/app/pull/15845/files#diff-fca4065dcdb1fb06adf96c5420beadf6R94

Reply:
That's a different hook that only has one parameter: https://github.com/Wikia/app/blob/2fb5243/includes/parser/Parser.php#L1882

Reply:
That's a different hook call, and due to the way it's checked, it still works as expected: https://github.com/Wikia/app/blob/2fb5243/includes/parser/Parser.php#L1882 However, it probably could explicitly return a value in this case since it's used in a conditional.

But, not returning a value here in `LinkerMakeExternalLink` https://github.com/Wikia/app/blob/2fb5243/includes/Linker.php#L1064 would break the functionality as we don't want to overwrite the output and stop the call to `Html::rawElement`, we just want to change the value of `$url` before the `Html::rawElement` call happens.